apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -954,20 +968,25 @@ type perContainerHandler interface {
 	ConfigType() string
 }
 
-func (p *ProvisionerAPI) processEachContainer(args params.Entities, handler perContainerHandler) error {
-	env, hostMachine, canAccess, err := p.prepareContainerAccessEnvironment()
+func (api *ProvisionerAPI) processEachContainer(args params.Entities, handler perContainerHandler) error {
+	env, hostMachine, canAccess, err := api.prepareContainerAccessEnvironment()
 	if err != nil {
 		// Overall error
 		return errors.Trace(err)
 	}
 	_, err = hostMachine.InstanceId()
 	if errors.IsNotProvisioned(err) {
-		err = errors.NotProvisionedf("cannot prepare container %s config: host machine %q", handler.ConfigType(), hostMachine)
-		return err
+		return errors.NotProvisionedf("cannot prepare container %s config: host machine %q",
+			handler.ConfigType(), hostMachine)
 	} else if err != nil {
 		return errors.Trace(err)
 	}
 
+	policy, err := containerizer.NewBridgePolicy(env, api.st)
```

Comment:
This is now created once per batch of containers instead of being created each time inside `ProcessOneContainer`.